### PR TITLE
ModuleHelper - also uses LC_ALL to force language

### LINKED ISSUE #2715: xfconf fails to set double value when LC_NUMERIC is set to nb_NO.UTF-8
### Summary

In https://github.com/ansible-collections/community.general/pull/744 `LANGUAGE` is used to force `xfconf-query` to return doubles using the expected format. This fails when `LC_NUMERIC` is set. From the article linked to in https://github.com/ansible-collections/community.general/pull/744, it seems like setting `LANGUAGE` should override `LC_NUMERIC`, but that isn't actually the case.

The correct variable to use in this case is probably `LC_ALL`.

I've attached a terminal recording showing the results. You'll notice that in the first run, the `previous_value` is `0,200000`, while after setting `LC_ALL=C`, the `previous_value` becomes `0.200000` which matches the input and no change is needed.

I've also attached the test-play I used (with a `.txt` extension because github doesn't like `.yml`).

[recording.txt](https://github.com/ansible-collections/community.general/files/6597487/recording.txt)

[test_play.yml](https://github.com/ansible-collections/community.general/files/6597488/test_play.yml.txt)

### Issue Type

Bug Report

### Component Name

xfconf

### Ansible Version

```console (paste below)
$ ansible --version
ansible 2.10.5
  config file = /home/mortenlj/.ansible.cfg
  configured module search path = ['/home/mortenlj/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.9.5 (default, May 11 2021, 08:20:37) [GCC 10.3.0]
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
DEFAULT_GATHERING(/home/mortenlj/.ansible.cfg) = smart
DEFAULT_HOST_LIST(/home/mortenlj/.ansible.cfg) = ['/home/mortenlj/code/personal/ansible/hosts']
DEFAULT_VAULT_PASSWORD_FILE(/home/mortenlj/.ansible.cfg) = /home/mortenlj/.ansible/vault-pass
```

### OS / Environment

```
$ cat /etc/os-release 
NAME="Ubuntu"
VERSION="21.04 (Hirsute Hippo)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 21.04"
VERSION_ID="21.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute

$ apt list xfce4-settings
Listing... Done
xfce4-settings/hirsute,now 4.16.0-1ubuntu1 amd64 [installed,automatic]
```

### Steps to Reproduce

See summary for links.

### Expected Results

I expect the play to not "change" the xfconf property on every run, because the new value should match the already set value.

### Actual Results

See summary. I realise I only used `-vvv`, but I don't think the extra `v` would make it any clearer.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Also setting the variable `LC_ALL` when forcing locale

Fixes #2715 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/module_utils/mh/mixins/cmd.py

### CODE REVIEW

**Comment by aminvakil** [comment]
This needs a changelog fragment :)

**Comment by russoz** [comment]
hi @mortenlj  would you be able to test this new code with your scenario. I think it is a pretty straightforward one, but just to be on the safe side.

File: changelogs/fragments/2731-mh-cmd-locale.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - module_helper module utils - ``CmdMixin`` must also use ``LC_ALL`` to enforce locale choice (https://github.com/ansible-collections/community.general/issues/2715).
```
Comment (felixfontein) [inline changelogs/fragments/2731-mh-cmd-locale.yml:2]:
```suggestion
  - module_helper module utils - ``CmdMixin`` must also use ``LC_ALL`` to enforce locale choice (https://github.com/ansible-collections/community.general/pull/2731).
  - xfconf - also use ``LC_ALL`` to enforce locale choice (https://github.com/ansible-collections/community.general/issues/2715).
```

Comment (russoz) [reply changelogs/fragments/2731-mh-cmd-locale.yml:2]: In that case should we include `cpanm` and `snap` as well? they were also affected by this change

Comment (felixfontein) [reply changelogs/fragments/2731-mh-cmd-locale.yml:2]: Sounds good.

**Review by felixfontein — APPROVED** [review]

**Review by aminvakil — APPROVED** [review]
LGTM

**Comment by mortenlj** [comment]
> hi @mortenlj would you be able to test this new code with your scenario. I think it is a pretty straightforward one, but just to be on the safe side.

If you tell me how to run `ansible-playbook` with the code in this PR, I can do quick testing.

**Comment by aminvakil** [comment]
@mortenlj 
See [this document](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_contributing.html), but instead of `ansible-collections/community.general` use `russoz/community.general` and make sure you've checked out the proper branch (`2715-modhelper-locale`) before testing.

**Comment by mortenlj** [comment]
I've tested it locally, and this PR seems to solve the issue. :+1:

**Comment by felixfontein** [comment]
@russoz thanks for fixing this!
@aminvakil thanks for reviewing!
@mortenlj thanks for testing!

**Comment by russoz** [comment]
hi @mortenlj  thanks for testing! Thanks @aminvakil for reviewing and assisting on the setup.

**Pull request merged**